draft-fossati-tls-exported-attestation.md

```diff
@@ -72,38 +86,119 @@ There is a growing need to demonstrate to a remote party that cryptographic keys
+
+- The attestation evidence is specific to the authentication event and cannot be replayed across different TLS sessions.
+- The attestation remains tied to the cryptographic context of the TLS session.
+
+## Ensuring Compatibility with X.509 Certificate Validation
+
+The `cmw_attestation` extension does not modify or replace X.509 certificate validation mechanisms. It serves as an additional source of authentication data rather than altering the trust model of PKI-based authentication. Specifically:
+
+- Certificate validation (e.g., signature verification, revocation checks) MUST still be performed according to TLS {{!RFC8446}} and PKIX {{!RFC5280}}.
+- The attestation evidence carried in `cmw_attestation` MUST NOT be used as a substitute for X.509 certificate validation but can be used alongside standard certificate validation for additional security assurances.
+- Implementations MAY reject connections where the certificate is valid but the attestation evidence is missing or does not meet security policy.
+
+## Applicability to Client and Server Authentication
+
+The `cmw_attestation` extension is applicable to both client and server authentication in Exported Authenticator-based post-handshake authentication.
+
+### Client Authentication
+For client authentication, the server acts as the relying party and requests attestation from the client. The client may respond with either:
+
+- Evidence (Passport Model):
+  - The client generates Evidence and includes it in the cmw_attestation extension.
+  - The server forwards the Evidence to a Verifier for evaluation and waits for an Attestation Result.
+  - The server grants or denies access based on the Attestation Result.
+
+- Attestation Result (Background Check Model):
+  - The client sends Evidence to a Verifier beforehand.
+  - The Verifier issues an Attestation Result to the client.
+  - The client includes the Attestation Result in cmw_attestation and sends it to the server.
+  - The server verifies the Attestation Result directly, without needing a separate Verifier.
+
+### Server Authentication
+For server authentication, the client acts as the relying party and requests attestation from the server. The server may respond with either:
+
+- Evidence (Passport Model):
+  - The server includes Evidence in cmw_attestation.
+  - The client forwards the Evidence to a Verifier and waits for an Attestation Result.
+  - The client evaluates the attestation result from the Verifier and may choose to continue or terminate the TLS session based on the assessment. If the attestation result fails security policy validation, the client will terminate the TLS session.
+
+- Attestation Result (Background Check Model):
+  - The server sends Evidence to a Verifier beforehand.
+  - The Verifier issues an Attestation Result to the server.
+  - The server includes the Attestation Result in cmw_attestation.
+  - The client validates the Attestation Result directly without consulting an external Verifier.
```

Comment:
Could we collapse this and the same write-up from above, and use `relying party` and `attester`, and just say that each of the roles are fulfilled by one or both parties?

Reply:
Good point, fixed.